src/device-manager/WeaveDataManagementClient.cpp

```diff
@@ -988,27 +1013,31 @@ void WdmClient::ClientEventCallback(void * const aAppState, SubscriptionClient::
         err = WEAVE_ERROR_INCORRECT_STATE;
         break;
     case SubscriptionClient::kEvent_OnUpdateComplete:
-        if ((aInParam.mUpdateComplete.mReason == WEAVE_NO_ERROR) &&
+        VerifyOrExit(kOpState_FlushUpdate == savedOpState, err = WEAVE_ERROR_INCORRECT_STATE);
+        if ((aInParam.mUpdateComplete.mReason ==  WEAVE_NO_ERROR) &&
             (nl::Weave::Profiles::kWeaveProfile_Common == aInParam.mUpdateComplete.mStatusProfileId) &&
             (nl::Weave::Profiles::Common::kStatus_Success == aInParam.mUpdateComplete.mStatusCode))
         {
-            WeaveLogDetail(DataManagement, "Update: path result: success");
+            WeaveLogDetail(DataManagement, "Update: path result: success, tdh %" PRIu16 ", ph %" PRIu16 ".", aInParam.mUpdateComplete.mTraitDataHandle, aInParam.mUpdateComplete.mPropertyPathHandle);
         }
         else
         {
-            WeaveLogDetail(DataManagement, "Update: path failed: %s, %s, tdh %" PRIu16 ", will %sretry, discard failed change",
+            WeaveLogDetail(DataManagement, "Update: path failed: %s, %s, tdh %" PRIu16 ", ph %" PRIu16 ",will %s retry, discard failed change",
```

Comment:
The "will %s retry" formatting section will lead to a double space between words. Consider just subbing the whole string? e.g.

"%s" -> "will retry" : "will not retry"

Reply:
yep